pilot/pkg/xds/endpoint_builder.go

```diff
@@ -132,6 +133,10 @@ func (b EndpointBuilder) Key() string {
 		hash.Write(b.failoverPriorityLabels)
 		hash.Write(Separator)
 	}
+	if b.service.Attributes.NodeLocal {
+		hash.Write([]byte(b.proxy.Labels[label.LabelHostname]))
```

Comment:
```suggestion
		hash.Write([]byte(b.proxy.NodeName))
```

Once other suggestions are done